### PR TITLE
added radix to parseInt() calls

### PR DESCRIPTION
**Pull request by mserranom**
## Description

Adds the `radix` parameter to `parseInt()` calls as suggested in https://github.com/overleaf/overleaf/pull/805#discussion_r510742503

### CODE REVIEW

**Review by jdleesmiller — COMMENTED** [review]
Good cleanup... but I think this code may need deeper changes. I can't find any situation in which `parseInt` will throw an error, looking at the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt), so all the try-catch stuff isn't really adding anything. I think it will probably make sense to use the helper you suggested in #805 here.

**Comment by mserranom** [comment]
Added a `parseIntOrFail()` function and (for consistency) updated `parse()` to throw an error when the JSON content is invalid (https://github.com/overleaf/overleaf/pull/806/commits/b162632510d868820ac915890c140940114f1380)

 [I wrote the function using the Coffeescript console](https://coffeescript.org/#try:parseIntOrFail%20%3D%20(value)-%3E%0A%09parsedValue%20%3D%20parseInt(value%2C%2010)%0A%09if%20isNaN(parsedValue)%0A%09%09throw%20new%20Error(%22'%23%7Bvalue%7D'%20is%20an%20invalid%20integer%22)%0A%09return%20parsedValue%0A%0Aconsole.log('testing...')%0A%0Aconsole.assert(5%20%3D%3D%20parseIntOrFail('5'))%0Aconsole.assert(5%20%3D%3D%20parseIntOrFail(5))%0A%0Atry%20%0A%09parseIntOrFail(false)%0A%09console.error('expected%20exception%20wasnt%20thrown')%0Acatch%20err%20then%20console.assert(err.message%20%3D%3D%20%22'false'%20is%20an%20invalid%20integer%22)%0A%0Atry%20%0A%09parseIntOrFail(%22something%22)%0A%09console.error('expected%20exception%20wasnt%20thrown')%0Acatch%20err%20then%20console.assert(err.message%20%3D%3D%20%22'something'%20is%20an%20invalid%20integer%22)%0A%0Atry%20%0A%09parseIntOrFail(%7Bx%3A%205%7D)%0A%09console.error('expected%20exception%20wasnt%20thrown')%0Acatch%20err%20then%20console.assert(err.message%20%3D%3D%20%22'%5Bobject%20Object%5D'%20is%20an%20invalid%20integer%22)%0A%0A%0Aconsole.log('testing%20ended'))

File: settings.coffee [hunk 1]
```diff
@@ -213,7 +217,7 @@ settings =
 		collaborators: -1
 		dropbox: true
 		versioning: true
-		compileTimeout: parseInt(process.env["COMPILE_TIMEOUT"] or 180, 10)
+		compileTimeout: parseIntOrFail(process.env["COMPILE_TIMEOUT"] or 180, 10)
```
Comment (jdleesmiller) [inline settings.coffee:220]: It looks like the radix on these calls isn't needed now?

**Review by jdleesmiller — COMMENTED** [review]
Approach looks good. Just the radixes to tidy up I think.

**Comment by mserranom** [comment]
Removed radix in https://github.com/overleaf/overleaf/pull/806/commits/b48a8400e461b2cf191afa7ad08bf4062cbf89e5. Will squash-merge the changes.

**Review by jdleesmiller — APPROVED** [review]
LGTM 👍

**Pull request merged**